### ISSUE TITLE
build problem on next with clang

### ISSUE THREAD

**Issue by larkost** [rethinkdb/rethinkdb#4895]
I am getting a build problem on next (1cd2122) on my local system (MacOS/clang), and it looks like it is a side-effect from @Tryneus's recent merge removing protobuffs:

```
src/rdb_protocol/response.cc:58:22: error: comparison of constant -1 with expression of type 'const Response::ResponseType' (aka 'const Response_ResponseType') is always true [-Werror,-Wtautological-constant-out-of-range-compare]
    do { if (!(type_ != -1)) { do { report_fatal_error("src/rdb_protocol/response.cc", 58, "Guarantee" " failed: [" "type_ != -1" "] "); (raise(5)); } while (0); } } while (0);
               ~~~~~ ^  ~~
src/rdb_protocol/response.cc:63:22: error: comparison of constant -1 with expression of type 'const Response::ResponseType' (aka 'const Response_ResponseType') is always true [-Werror,-Wtautological-constant-out-of-range-compare]
    do { if (!(type_ != -1)) { do { report_fatal_error("src/rdb_protocol/response.cc", 63, "Guarantee" " failed: [" "type_ != -1" "] "); (raise(5)); } while (0); } } while (0);
               ~~~~~ ^  ~~
src/rdb_protocol/response.cc:68:22: error: comparison of constant -1 with expression of type 'const Response::ResponseType' (aka 'const Response_ResponseType') is always true [-Werror,-Wtautological-constant-out-of-range-compare]
    do { if (!(type_ != -1)) { do { report_fatal_error("src/rdb_protocol/response.cc", 68, "Guarantee" " failed: [" "type_ != -1" "] "); (raise(5)); } while (0); } } while (0);
               ~~~~~ ^  ~~
src/rdb_protocol/response.cc:73:22: error: comparison of constant -1 with expression of type 'const Response::ResponseType' (aka 'const Response_ResponseType') is always true [-Werror,-Wtautological-constant-out-of-range-compare]
    do { if (!(type_ != -1)) { do { report_fatal_error("src/rdb_protocol/response.cc", 73, "Guarantee" " failed: [" "type_ != -1" "] "); (raise(5)); } while (0); } } while (0);
               ~~~~~ ^  ~~
src/rdb_protocol/response.cc:78:22: error: comparison of constant -1 with expression of type 'const Response::ResponseType' (aka 'const Response_ResponseType') is always true [-Werror,-Wtautological-constant-out-of-range-compare]
    do { if (!(type_ != -1)) { do { report_fatal_error("src/rdb_protocol/response.cc", 78, "Guarantee" " failed: [" "type_ != -1" "] "); (raise(5)); } while (0); } } while (0);
               ~~~~~ ^  ~~
src/rdb_protocol/response.cc:83:22: error: comparison of constant -1 with expression of type 'const Response::ResponseType' (aka 'const Response_ResponseType') is always true [-Werror,-Wtautological-constant-out-of-range-compare]
    do { if (!(type_ != -1)) { do { report_fatal_error("src/rdb_protocol/response.cc", 83, "Guarantee" " failed: [" "type_ != -1" "] "); (raise(5)); } while (0); } } while (0);
               ~~~~~ ^  ~~
6 errors generated.
```

**Comment by danielmewes** [comment]
The `ResponseType` type seems to be an enum, and it doesn't have any negative values. So I guess it depends on the compiler whether it uses a signed or unsigned type to represent it.

@Tryneus Do you remember why you are guaranteeing that it's not `-1`? Under which circumstances would a `-1` show up here?

**Comment by Tryneus** [comment]
This should probably be doing a `static_cast`.  `-1` is used here to indicate an uninitialized response type, which should only happen if we try to send a response that was not properly filled (thus the guarantee).

**Comment by danielmewes** [comment]
This is still not really clear to me. What would cause an uninitialized type to be set to `-1`?

**Comment by Tryneus** [comment]
The type is set to -1 in the `response_t::clear()` function, which is called in a few places, including the `response_t` constructor.

**Comment by danielmewes** [comment]
Fix in CR 3270

**Comment by danielmewes** [comment]
Fixed in `next` 7e06b3a
